tools/istio-iptables/pkg/dependencies/implementation_linux.go

```diff
@@ -82,72 +61,73 @@ func (r *RealDependencies) execute(cmd string, ignoreErrors bool, stdin io.Reade
 	return err
 }
 
-func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, stdin io.ReadSeeker, args ...string) error {
-	if r.CNIMode && r.HostNSEnterExec {
-		originalCmd := cmd
-		cmd = constants.NSENTER
-		args = append([]string{fmt.Sprintf("--net=%v", r.NetworkNamespace), "--", originalCmd}, args...)
-	}
-	log.Infof("Running command: %s %s", cmd, strings.Join(args, " "))
-
-	var stdout, stderr *bytes.Buffer
-	var err error
-	var nsContainer ns.NetNS
-
-	if r.CNIMode && !r.HostNSEnterExec {
-		nsContainer, err = ns.GetNS(r.NetworkNamespace)
+func (r *RealDependencies) runCommand(c *exec.Cmd) error {
+	if r.CNIMode {
+		n, err := ns.GetNS(r.NetworkNamespace)
 		if err != nil {
 			return err
 		}
-		defer nsContainer.Close()
-	}
-	o := backoff.Option{
-		InitialInterval: 100 * time.Millisecond,
-		MaxInterval:     2 * time.Second,
+		defer n.Close()
+
+		return n.Do(func(ns.NetNS) error {
+			return c.Run()
+		})
 	}
-	b := backoff.NewExponentialBackOff(o)
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-	backoffError := b.RetryWithContext(ctx, func() error {
```

Comment:
no more retry?

Reply:
The retry was a poor replacement for `--wait`. It was only retrying lock acquire failure; `--wait` does similar but much better. In the worst case where `--wait` fails, the container runtime will retry anyways

Reply:
yeah

Reply:
I assume you don't support older versions than 1.4.20 https://github.com/kubernetes/kubernetes/blob/ba0505319e11f9b9cae458b41b8adfb73b97a312/pkg/util/iptables/iptables.go#L174

Reply:
We only use `iptables-restore`, I think 1.4 was when it was added to `iptables`; 1.6.2 added it to `iptables-restore`. Anyhow, we have a minimum GLIBC version that is much newer than iptables 1.4, so unless someone has a really bizarre OS we should never see 1.4

Reply:
Even Centos7 we dropped, which has 1.4.21